android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java

```diff
@@ -5,453 +5,57 @@
+        new TabLayoutMediator(mTabLayout, mViewPager,
+                (tab, position) -> tab.setText(mViewPagerAdapter.getPageTitle(position)))
+                .attach();
     }
 
     public void callAnotherApproveDialog() {
-        if (!hasPendingTx() || mWalletModel == null) {
-            return;
-        }
-        ApproveTxBottomSheetDialogFragment approveTxBottomSheetDialogFragment =
-                ApproveTxBottomSheetDialogFragment.newInstance(mCurrentPendingTx,
-                        mWalletModel.getCryptoModel()
-                                .getPendingTxHelper()
-                                .getAccountNameForTransaction(mCurrentPendingTx));
-        approveTxBottomSheetDialogFragment.setApprovedTxObserver(this);
-        approveTxBottomSheetDialogFragment.show(
-                getParentFragmentManager(), ApproveTxBottomSheetDialogFragment.TAG_FRAGMENT);
-    }
-
-    private void showEditVisibleDialog() {
-        if (mNetworkInfo == null) {
-            return;
-        }
-
-        EditVisibleAssetsBottomSheetDialogFragment bottomSheetDialogFragment =
-                EditVisibleAssetsBottomSheetDialogFragment.newInstance(
-                        WalletCoinAdapter.AdapterType.EDIT_VISIBLE_ASSETS_LIST, false);
-
-        bottomSheetDialogFragment.setSelectedNetwork(mNetworkInfo);
-        bottomSheetDialogFragment.setDismissListener(
-                new EditVisibleAssetsBottomSheetDialogFragment.DismissListener() {
-                    @Override
-                    public void onDismiss(boolean isAssetsListChanged) {
-                        if (isAssetsListChanged) {
-                            updatePortfolioGetPendingTx();
-                        }
-                    }
-                });
-
-        bottomSheetDialogFragment.show(
-                getFragmentManager(), EditVisibleAssetsBottomSheetDialogFragment.TAG_FRAGMENT);
-    }
-
-    private void updateNextPendingTx() {
-        if (mCurrentPendingTx != null) {
-            for (TransactionInfo info : mPendingTxs) {
-                if (!mCurrentPendingTx.id.equals(info.id)
-                        && info.txStatus == TransactionStatus.UNAPPROVED) {
-                    mCurrentPendingTx = info;
-                    return;
-                }
-            }
-            mCurrentPendingTx = null;
-        } else if (mPendingTxs.size() > 0) {
-            mCurrentPendingTx = mPendingTxs.get(0);
-        }
-    }
-
-    private boolean hasPendingTx() {
-        return mCurrentPendingTx != null;
-    }
-
-    private void updatePendingTxNotification() {
-        Activity activity = getActivity();
-        if (activity instanceof BraveWalletActivity)
-            ((BraveWalletActivity) activity).showPendingTxNotification(hasPendingTx());
-    }
-
-    private NetworkModel getNetworkModel() {
-        return mWalletModel.getCryptoModel().getNetworkModel();
+        mViewPagerAdapter.getAssetsFragment().callAnotherApproveDialog();
```

Comment:
Do we need to move the pending transactions logic in a nested asset fragment? Better to keep it in the parent `Portfolio` fragment.